dna/__main__.py

```diff
@@ -154,29 +187,20 @@ def evaluate_handler(
     for problem_name in getattr(arguments, 'problem'):
         if arguments.verbose:
             print('{} {} {}'.format(model_name, problem_name, run_id))
-        problem = problem_resolver(problem_name)
-        if problem_name == 'rank':  # todo fix this hack to allow problem args
-            k = getattr(arguments, 'k')
-            train_predictions, test_predictions, train_scores, test_scores, timings = problem.run(
-                train_data, test_data, model, k, getattr(arguments, 'scores'), model_config=model_config,
-                re_fit_model=False, verbose=arguments.verbose, output_dir=model_output_dir
-            )
-        else:
-            train_predictions, test_predictions, train_scores, test_scores, timings = problem.run(
-                train_data, test_data, model, model_config=model_config, re_fit_model=False,
-                verbose=arguments.verbose, output_dir=model_output_dir
-            )
+        problem = problem_resolver(problem_name, arguments)
+        evaluate_result = evaluate(
```

Comment:
Ddi we not implement the changes for the `out-of-training-set-pipelines`?  Is that going to be a new PR?

Reply:
I just did not get to it last night. Let's do a separate PR. This one is big enough.